guide/interactions/select-menus.md

```diff
@@ -0,0 +1,308 @@
+
+```js {1,26-30,32}
+const { MessageActionRow, MessageSelectMenu, MessageEmbed } = require('discord.js');
+
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const row = new MessageActionRow()
+			.addComponents(
+				new MessageSelectMenu()
+					.setCustomID('select')
+					.setPlaceholder('Nothing selected')
+					.addOptions([
+						{
+							label: 'Select me',
+							description: 'This is a description',
+							value: 'first_option',
+						},
+						{
+							label: 'You can select me too',
+							description: 'This is also a description',
+							value: 'second_option',
+						},
+					]),
+			);
+
+		const embed = new MessageEmbed()
+			.setColor('#0099ff')
+			.setTitle('Some title')
+			.setURL('https://discord.js.org/')
+			.setDescription('Some description here');
+
+		await interaction.reply({ content: 'Pong!', ephemeral: true, embeds: [embed], components: [row] });
+	}
+});
+```
+
+Restart your bot and then send the command to a channel your bot has access to. If all goes well, you should see something like this:
+
+<!--- vue-discord-message doesn't yet have support for inline replies/interactions/ephemeral messages/components -->
+<div is="discord-messages">
+	<discord-message profile="bot">
+		Pong! (ephemeral + components)
+		<discord-embed
+			slot="embeds"
+			color="#0099ff"
+			title="Some title"
+			url="https://discord.js.org/"
+		>
+			Some description here
+		</discord-embed>
+	</discord-message>
+</div>
+
+Additionally, if you don't want to construct an `ActionRow` every time, you can also pass an array of arrays containing components like this:
+
+```js {7-21,23}
+const { MessageActionRow, MessageSelectMenu, MessageEmbed } = require('discord.js');
+
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const menu = new MessageSelectMenu()
+			.setCustomID('select')
+			.setPlaceholder('Nothing selected')
+			.addOptions([
+				{
+					label: 'Select me',
+					description: 'This is a description',
+					value: 'first_selection',
+				},
+				{
+					label: 'You can select me too',
+					description: 'This is also a description',
+					value: 'second_selection',
+				},
+			]);
+
+		await interaction.reply({ content: 'Pong!', ephemeral: true, embeds: [embed], components: [[menu]] });
+	}
+});
+```
+
+Now you know all there is to building and sending a `SelectMenu`! Let's move on to how to receive selected options!
+
+## Receiving select menu options
+
+## Receiving Select menus
+
+To receive a `SelectMenuInteractionInteraction`, simply attach an event listener to your client and also use the `Interaction#isSelectMenu()` typeguard to make sure you only receive select menus:
+
+```js {2}
+client.on('interaction', interaction => {
+	if (!interaction.isSelectMenu()) return;
+	if (interaction.componentType !== 'SELECT_MENU') return;
```

Comment:
```suggestion
	if (!interaction.isSelectMenu()) return;
```
Component type is checked in the first line - I assume this was leftover from when only `isMessageComponent()` was available.

Reply:
adressed in the recent commits 👍🏼